retry.go

```diff
@@ -0,0 +1,95 @@
+/*
+ * Minio Go Library for Amazon S3 Compatible Cloud Storage (C) 2015, 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package minio
+
+import (
+	"math"
+	"time"
+)
+
+// maxRetries is the maximum number of retries before stopping.
+var maxRetries = 5
+
+// ErrMaxRetriesReached - Maximum retries reached.
+func ErrMaxRetriesReached(message string) error {
```

Comment:
This error should never be exposed. Users should only see the real error such as connection refused or host unreachable.